src/models/partial_tx.ts

```diff
@@ -0,0 +1,440 @@
+  token: string;
+  value: number;
+  address: string;
+
+  constructor(hash: string, index: number, token: string, value: number, address: string) {
+    super(hash, index);
+    this.token = token;
+    this.value = value;
+    this.address = address;
+  }
+
+  /**
+   * Return an object with the relevant input data
+   *
+   * @return {InputData}
+   * @memberof ProposalInput
+   * @inner
+   */
+  toData() {
+    return {
+      tx_id: this.hash,
+      index: this.index,
+      data: Buffer.from([]),
+      address: this.address,
+    };
+  }
+}
+
+export class ProposalOutput extends Output {
+  token: string;
+  isChange: Boolean;
+
+  /**
+   * We do not set tokenData because the token array is not yet formed
+   */
+  constructor(value: number, script: Buffer, token: string, isChange: Boolean) {
+    super(value, script);
+    this.token = token;
+    this.isChange = isChange;
+  }
+
+  setTokenData(tokenData: number) {
+    this.tokenData = tokenData;
+  }
+
+  /**
+   * Return an object with the relevant output data
+   *
+   * @throws {UnsupportedScriptError} Script must be P2SH or P2PKH
+   *
+   * @return {OutputData}
+   * @memberof ProposalOutput
+   * @inner
+   */
+  toData(tokenData: number, network: Network) {
+    const script = this.parseScript(network);
+    if (!(script instanceof P2PKH || script instanceof P2SH)) {
+      throw new UnsupportedScriptError('Unsupported script type');
+    }
+
+    this.setTokenData(tokenData);
+
+    const data: OutputData = {
+      type: script.getType(),
+      value: this.value,
+      tokenData: tokenData,
+      address: script.address.base58,
+    };
+    if (script.timelock) {
+      data.timelock = script.timelock;
+    }
+
+    return data
+  }
+}
+
+/**
+ * This class purpose is serialization/deserialization of signatures from a MultiSig participant
+ * The structure of the serialized signature string is:
+ * "<pubkey>|<index>:<signature>|<index>:<signature>|..."
+ *
+ * The `pubkey` is required so we can identify the original signer (and his position on the redeemScript)
+ * The `<index>:<signature>` pair is the input index and the signature for that input.
+ * The signature is formatted to DER and hex encoded.
+ *
+ * With this information we will be able to encode the signatures for all inputs on one string.
+ * It also has all information needed to assemble the input data if you have enough participants' P2SHSignature serialized signatures.
+ */
+export class PartialTx {
+  inputs: ProposalInput[];
+  outputs: ProposalOutput[];
+  network: Network;
+
+  constructor(network?: Network) {
+    this.inputs = [];
+    this.outputs = [];
+    this.network = network || new Network('mainnet');
+  }
+
+  /**
+   * Convert the PartialTx into a complete TxData ready to be signed or serialized.
+   *
+   * @throws {UnsupportedScriptError} All output scripts must be P2SH or P2PKH
+   *
+   * @return {TxData}
+   * @memberof PartialTx
+   * @inner
+   */
+  getTxData() {
```

Comment:
We use the `getTxData` to use the `transaction.prepareData` which signs the owned inputs and then we can add any other input data before using the `createTxFromData` to create the complete transaction with input data from multiple wallets.

The `getTx` method is simply a helper to get the "current state" of the atomic-swap as a transaction.

Reply:
I'm wondering how much more work would be not to use the old methods. Every new method/class I create in the lib, I try not to use the methods in the root files because they are old and should be deprecated. Should we create a new method just to sign the inputs from a `Transaction` object? The you wouldn't need to use object data anymore.

Reply:
I think we should have a method `signTx(privkey)` on the `Transaction` model, this would make it very simple to manage transactions.
But I think this should be it's own PR, followed by a refactor to use it where needed.